File: MiniDOMSIP/Config/Linee/LineaEsterna.cs
using System;
using DMD;
using DMD.XML;

namespace minidom.PBX
{
    // Implements Xml.Serialization.IXmlSerializable
    public class LineaEsterna 
        : IDMDXMLSerializable, IComparable
    {
        public string Nome;
        public string Prefisso;
        public int Ordine;
        public string Server;

        public LineaEsterna()
        {
            DMDObject.IncreaseCounter(this);
            Nome = "";
            Prefisso = "";
            Ordine = 0;
            Server = "";
        }

        public LineaEsterna(string nome, string prefisso) : this()
        {
            Nome = DMD.Strings.Trim(nome);
            Prefisso = DMD.Strings.Trim(prefisso);
        }

        public override string ToString()
        {
            return Nome + " (" + Prefisso + ")";
        }


        // Private Function GetSchema() As Xml.Schema.XmlSchema Implements Xml.Serialization.IXmlSerializable.GetSchema
        // Return Nothing
        // End Function

        // Private Sub ReadXml(reader As Xml.XmlReader) Implements Xml.Serialization.IXmlSerializable.ReadXml
        // Me.Nome = reader.GetAttribute("nome")
        // Me.Prefisso = reader.GetAttribute("prefisso")
        // reader.Read()
        // End Sub

        // Public Sub WriteXml(writer As Xml.XmlWriter) Implements Xml.Serialization.IXmlSerializable.WriteXml
        // writer.WriteAttributeString("nome", Me.Nome)
        // writer.WriteAttributeString("prefisso", Me.Prefisso)
        // End Sub

        protected virtual void SetFieldInternal(string fieldName, object fieldValue)
        {
            switch (fieldName ?? "")
            {
                case "Nome":
                    {
                        Nome = DMD.XML.Utils.Serializer.DeserializeString(fieldValue);
                        break;
                    }

                case "Prefisso":
                    {
                        Prefisso = DMD.XML.Utils.Serializer.DeserializeString(fieldValue);
                        break;
                    }

                case "Ordine":
                    {
                        Ordine = (int)DMD.XML.Utils.Serializer.DeserializeInteger(fieldValue);
                        break;
                    }

                case "Server":
                    {
                        Server = DMD.XML.Utils.Serializer.DeserializeString(fieldValue);
                        break;
                    }
            }
        }

        protected virtual void XMLSerialize(XMLWriter writer)
        {
            writer.WriteAttribute("Nome", Nome);
            writer.WriteAttribute("Prefisso", Prefisso);
            writer.WriteAttribute("Ordine", Ordine);
            writer.WriteAttribute("Server", Server);
        }

        public int CompareTo(LineaEsterna linea)
        {
            int ret = Ordine.CompareTo(linea.Ordine);
            if (ret == 0)
                ret = DMD.Strings.Compare(Nome, linea.Nome, true);
            if (ret == 0)
                ret = DMD.Strings.Compare(Prefisso, linea.Prefisso, true);
            return ret;
        }

        int IComparable.CompareTo(object obj)
        {
            return CompareTo((LineaEsterna)obj);
        }

        void IDMDXMLSerializable.XMLSerialize(XMLWriter writer)
        {
            this.XMLSerialize(writer);
        }

        void IDMDXMLSerializable.SetFieldInternal(string fieldName, object fieldValue)
        {
            this.SetFieldInternal(fieldName, fieldValue);
        }

        ~LineaEsterna()
        {
            DMDObject.DecreaseCounter(this);
        }
    }
}

File: MiniDOMSIP/Config/LineaEsterna.vb
Imports minidom
Imports minidom.Sistema

Public Class LineaEsterna
    'Implements Xml.Serialization.IXmlSerializable
    Implements XML.IDMDXMLSerializable, IComparable

    Public Nome As String
    Public Prefisso As String
    Public Ordine As Integer
    Public Server As String

    Public Sub New()
        DMDObject.IncreaseCounter(Me)
        Me.Nome = ""
        Me.Prefisso = ""
        Me.Ordine = 0
        Me.Server = ""
    End Sub

    Public Sub New(ByVal nome As String, ByVal prefisso As String)
        Me.New
        Me.Nome = Trim(nome)
        Me.Prefisso = Trim(prefisso)
    End Sub

    Public Overrides Function ToString() As String
        Return Me.Nome & " (" & Me.Prefisso & ")"
    End Function


    'Private Function GetSchema() As Xml.Schema.XmlSchema Implements Xml.Serialization.IXmlSerializable.GetSchema
    '    Return Nothing
    'End Function

    'Private Sub ReadXml(reader As Xml.XmlReader) Implements Xml.Serialization.IXmlSerializable.ReadXml
    '    Me.Nome = reader.GetAttribute("nome")
    '    Me.Prefisso = reader.GetAttribute("prefisso")
    '    reader.Read()
    'End Sub

    'Public Sub WriteXml(writer As Xml.XmlWriter) Implements Xml.Serialization.IXmlSerializable.WriteXml
    '    writer.WriteAttributeString("nome", Me.Nome)
    '    writer.WriteAttributeString("prefisso", Me.Prefisso)
    'End Sub

    Protected Overridable Sub SetFieldInternal(fieldName As String, fieldValue As Object) Implements XML.IDMDXMLSerializable.SetFieldInternal
        Select Case fieldName
            Case "Nome" : Me.Nome = XML.Utils.Serializer.DeserializeString(fieldValue)
            Case "Prefisso" : Me.Prefisso = XML.Utils.Serializer.DeserializeString(fieldValue)
            Case "Ordine" : Me.Ordine = CInt(XML.Utils.Serializer.DeserializeInteger(fieldValue))
            Case "Server" : Me.Server = XML.Utils.Serializer.DeserializeString(fieldValue)
        End Select
    End Sub

    Protected Overridable Sub XMLSerialize(writer As XML.XMLWriter) Implements XML.IDMDXMLSerializable.XMLSerialize
        writer.WriteAttribute("Nome", Me.Nome)
        writer.WriteAttribute("Prefisso", Me.Prefisso)
        writer.WriteAttribute("Ordine", Me.Ordine)
        writer.WriteAttribute("Server", Me.Server)
    End Sub

    Public Function CompareTo(ByVal linea As LineaEsterna) As Integer
        Dim ret As Integer = Me.Ordine.CompareTo(linea.Ordine)
        If (ret = 0) Then ret = Strings.Compare(Me.Nome, linea.Nome)
        If (ret = 0) Then ret = Strings.Compare(Me.Prefisso, linea.Prefisso)
        Return ret
    End Function

    Private Function CompareTo1(obj As Object) As Integer Implements IComparable.CompareTo
        Return Me.CompareTo(DirectCast(obj, LineaEsterna))
    End Function

    Protected Overrides Sub Finalize()
        MyBase.Finalize()
        DMDObject.DecreaseCounter(Me)
    End Sub
End Class
